dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/parser/TaskOutputParameterParser.java

```diff
@@ -0,0 +1,107 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dolphinscheduler.plugin.task.api.parser;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Used to parse ${setValue()} and #{setValue()} from given lines.
+ */
+@Slf4j
+@NotThreadSafe
+public class TaskOutputParameterParser {
+
+    private final Map<String, String> taskOutputParams = new HashMap<>();
+
+    private List<String> currentTaskOutputParam;
+
+    public void appendParseLog(String log) {
+        if (log == null) {
+            return;
+        }
+
+        if (currentTaskOutputParam != null) {
+            // continue to parse the rest of line
+            int i = log.indexOf(")}");
+            if (i == -1) {
+                // the end of var pool not found
+                currentTaskOutputParam.add(log);
+            } else {
+                // the end of var pool found
+                currentTaskOutputParam.add(log.substring(0, i + 2));
+                Pair<String, String> keyValue = parseOutputParam(String.join("\n", currentTaskOutputParam));
+                if (keyValue.getKey() != null && keyValue.getValue() != null) {
+                    taskOutputParams.put(keyValue.getKey(), keyValue.getValue());
+                }
+                currentTaskOutputParam = null;
+                // continue to parse the rest of line
+                if (i + 2 != log.length()) {
+                    appendParseLog(log.substring(i + 2));
+                }
+            }
+            return;
+        }
+
+        int indexOfVarPoolBegin = log.indexOf("${setValue(");
+        if (indexOfVarPoolBegin == -1) {
+            indexOfVarPoolBegin = log.indexOf("#{setValue(");
+        }
+        if (indexOfVarPoolBegin == -1) {
+            return;
+        }
+        currentTaskOutputParam = new ArrayList<>();
+        appendParseLog(log.substring(indexOfVarPoolBegin));
+    }
+
+    public Map<String, String> getTaskOutputParams() {
+        return taskOutputParams;
+    }
+
+    // #{setValue(xx=xx)}
+    protected Pair<String, String> parseOutputParam(String outputParam) {
+        if (StringUtils.isEmpty(outputParam)) {
+            log.info("The task output param is empty");
+            return ImmutablePair.nullPair();
+        }
+        if ((!outputParam.startsWith("${setValue(") && !outputParam.startsWith("#{setValue("))
+                || !outputParam.endsWith(")}")) {
```

Comment:
Have to deal with output parameters in k8s task which end with `dsVal}`

Reply:
Why do we use two styles? is there any reason?

Reply:
ref https://github.com/apache/dolphinscheduler/pull/14934#discussion_r1336739386

Reply:
I've just noticed that you use `indexOf` in current implementation of `TaskOutputParameterParser`, if you think there's no performance issue, we can keep the same way with k8s task.

Reply:
We need to use indexOf rather than `startWith` or `endWith` since the outputParam might exist in any way.

Reply:
And use `)}` is enough, we don't need to import a new grammatical

Reply:
Ok, we can use the uniform way, BTW, you can update doc in `context.md` in which specify output parameter usage of k8s task.